reconcile/typed_queries/status_board.py

```diff
@@ -27,19 +27,24 @@ def get_selected_app_names(
     global_selectors: Iterable[str],
     product: StatusBoardProductV1,
 ) -> set[str]:
-    selected_app_names: set[str] = {
-        namespace.app.name for namespace in product.product_environment.namespaces or []
-    }
+    selected_app_names: set[str] = set()
 
     apps: dict[str, Any] = {"apps": []}
     for namespace in product.product_environment.namespaces or []:
-        apps["apps"].append(namespace.app.dict(by_alias=True))
+        prefix = ""
+        if namespace.app.parent_app:
+            prefix = f"{namespace.app.parent_app.name}-"
+        selected_app_names.add(f"{prefix}{namespace.app.name}")
+        app = namespace.app.dict(by_alias=True)
+        app["name"] = f"{prefix}{namespace.app.name}"
+        apps["apps"].append(app)
```

Comment:
This will change `name` in `apps`, will this affect the result of selector usage https://github.com/app-sre/qontract-reconcile/pull/3886/files#diff-a660d41a807b3693327e4b178284222146e6a42507601df0e8bbcf6facb2c89aR56?

Reply:
I added another test case for this

Reply:
Nice! that `f"{prefix}{namespace.app.name}"` can be extracted to a `name` so we can see clearly the one added to `selected_app_names` is the same as `app["name"]`